summaries/analysis/Analyzer.py

```diff
@@ -207,3 +217,213 @@ def _get_tokens(self, text: str) -> List[str]:
         else:
             return [token.text for token in self.processor(text)]
 
+    def find_identity_samples(self,
+                              reference_text_column_name: str,
+                              summary_text_column_name: str,
+                              train_set: Optional[Union[List, Dataset]] = None,
+                              validation_set: Optional[Union[List, Dataset]] = None,
+                              test_set: Optional[Union[List, Dataset]] = None,
+                              comparison_method: str = "exact") -> None:
+        """
+        Function to determine the number of samples where the input and output is the same.
+        :param reference_text_column_name: Name of the column that contains the reference text, for a sample.
+            This assumes that each sample is structured as a Dict.
+        :param summary_text_column_name: Name of the column that contains the summary text.
+        :param train_set: List of samples, or split of Huggingface dataset, containing the training samples.
+            It is assumed that this is the largest split.
+        :param validation_set: List of samples, or split of Huggingface dataset, containing validation samples.
+        :param test_set: List of samples, or split of Huggingface dataset, containing test samples.
+        :param comparison_method: Which method to use to determine similarity. Currently only supports "exact",
+            but future versions could utilize approximate matchers to be more resilient to, e.g., Unicode issues.
+        :return: Nothing is returned, but numbers of affected samples are printed to console.
+        """
+        # FIXME: This currently will also register samples that are simply two empty strings, which is technically
+        #  already handled by self.is_either_text_empty().
+        if comparison_method not in valid_comparison_methods:
+            raise ValueError(f"Currently only the following comparison methods are supported: "
+                             f"{valid_comparison_methods}")
+
+        identity_sample_count = 0
+        for sample in train_set:
+            if sample[reference_text_column_name] == sample[summary_text_column_name]:
+                print(f"Found affected training sample with same reference and summary text:")
+                print(f"{sample[:self.print_cutoff_length]}")
+                identity_sample_count += 1
+        for sample in validation_set:
+            if sample[reference_text_column_name] == sample[summary_text_column_name]:
+                print(f"Found affected validation sample with same reference and summary text:")
+                print(f"{sample[:self.print_cutoff_length]}")
+                identity_sample_count += 1
+        for sample in test_set:
+            if sample[reference_text_column_name] == sample[summary_text_column_name]:
```

Comment:
I just realized that this *does* require the check like `if train_split:`, etc., given that the default is a `NoneType` object, which cannot be iterated over. In such a case, `for sample in train_set` would throw an uncaught error.